### DISCUSSION TITLE
Filter for Date of offer

### DISCUSSION THREAD

**Discussion by Nik-git1** [0xku/leetcode-compensation#44] [Ideas]
We can add this filter so that the user can get the idea for the current market situation, like most of the old offers wont make sense  now and tamper the stats. For design, we can go with a dual range selector going from 1 month to 2+ year option with 6 month(or 3) threshold of selector.
![slider](https://github.com/kuutsav/leetcode-compensation/assets/88608366/c4910718-b156-4c22-abf7-1a47baf0c2bd)

**Comment by 0xku** [comment]
There is an upper limit of 2k posts that we will process/show. After post processing filters this would come down to 1.2-1.5k at max. This hard limit is set keeping in mind recency and the architecture of the tool.
It's serving static content, there is no db and i want to be as lightweight as possible. Currently the overall assets and data amount to 1-1.5mb in total.

What I'm trying to say is that recency will never be a concern as it's built into the system as hard limits.

Instead of making any more UI changes, if you want to you can look at the role mapping and try to create a good sanitisation logic over there, it needs some manual effort from time to time to update the raw -> mapped_role mapping.
